### ISSUE TITLE
Simulation of multiple tests very slow when using multithreading

### ISSUE THREAD

**Issue by jonnykl** [SpinalHDL/SpinalHDL#470]
The simulation of the second (and following runs) of `doSim` is very slow when using verilator multithreading. The following code demonstrates the issue:

```scala
// generate some logic so verilator has some work to do
case class Foo(a: Int, b: Int) extends Component {
  val io = new Bundle {
    val output = out UInt(a bits)
  }

  var tmp = U(0, a bits)

  for (_ <- 0 until b) {
    val reg = Reg(UInt(a bits)) init(0)
    reg := reg + 1

    tmp = tmp + reg
  }

  io.output := tmp
}


object Foo {
  def main(args: Array[String]): Unit = {
    val cfg = SimConfig.addSimulatorFlag("--threads 3")
    val compiled = cfg.compile(Foo(a=2048, b=10))

    def test() = {
      compiled.doSim(seed=0) { dut =>
        dut.clockDomain.forkStimulus(10)

        // just wait for 2000 clock cycles
        dut.clockDomain.waitSampling(2000)
      }
    }

    test()  // fast, < 50 ms
    test()  // very slow, ~ 15 s
  }
}
```

This is not only a problem when running `doSim` multiple times in one test, but also when running two tests using multithreading in sbt with `testOnly TestA TestB` as all tests are run in a single process. Then the simulation in TestA would be fast and the simulation in TestB would be slow.

**Comment by jonnykl** [comment]
Seems to be fixed in v1.6.0 (tested previously with v1.4.3)

**Comment by Dolu1990** [comment]
Hmm, maybe it was related to the cpu affinity fix, i don't know XD

Thanks anyway ^^
